components/postgresql/sources/new-row/new-row.mjs

```diff
@@ -0,0 +1,60 @@
+import common from "../common.mjs";
+
+export default {
+  ...common,
+  name: "New Row",
+  key: "postgresql-new-row",
+  description: "Emit new event when a new row is added to a table",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    table: {
+      propDefinition: [
+        common.props.postgresql,
+        "table",
+      ],
+    },
+    column: {
+      propDefinition: [
+        common.props.postgresql,
+        "column",
+        (c) => ({
+          table: c.table,
+        }),
+      ],
+      optional: true,
+    },
+  },
+  hooks: {
+    /** If column prop is left blank, get the table's primary key to use
+     * for ordering and deduping. */
+    async deploy() {
+      const column = this.column
+        ? this.column
+        : await this.postgresql.getPrimaryKey(this.table);
+      this._setColumn(column);
+    },
+  },
+  methods: {
+    ...common.methods,
+    _getColumn() {
+      return this.db.get("column");
+    },
+    _setColumn(column) {
+      this.db.set("column", column);
+    },
+    generateMeta(row, column) {
+      return {
+        id: row[column],
+        summary: "New Row Added",
+        ts: Date.now(),
+      };
+    },
+  },
+  async run() {
+    const column = this._getColumn();
+    await this.newRows(this.table, column);
+  },
```

Comment:
There is an issue with this one because if I have a table with these columns `(id, name)` and I select the column `name` in the setup config, that means the rows will be order by `name`. So then let's supose I'm adding the first row as
```sql
INSERT INTO test_table (id, name) values (1, 'George');
```
and right after the insert I run the source, so in this case the `lastResult` values will be `George`. So then if I need to insert the next row like this
```sql
INSERT INTO test_table (id, name) values (2, 'Andrew');
```
and then the trigger gets called. It won't display this new record since the latest `name` inserted was `George`  and according to the sql statement here in `getRows` function
```sql
...
WHERE " + column + ">" + lastResult
...
```
it won't show the row with `Andrew` in the `name`.

So I think a practical solution for this is when we are deploying we should validate if the colum is `unique` if not, throws with error advicing this to the user. In that way the data would be more consistent when it comes to get new inserted rows. I was testing this on Zapier and it's the same aproach, they seem to not allow to select a column that is not unique before proceed with next step.

Another solution would be to make the user know that the `column` property should be datetime type the same aproach that you took for  `new-or-updated-row.mjs` in the property description.

Let me see what are your thoughts on this.

Reply:
I updated `new-row.mjs` so that it doesn't use `lastResult`. It will just use the selected column to dedupe. I also added in a method to check that the column provided is unique (checks that it contains no duplicate values) which is used in sources `new-or-updated-row.mjs`, `new-row.mjs`, and `new-row-custom-query.mjs`.